### PR TITLE
Simplify system logs creation

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Not for changelog

Simplify system logs creation. Now SYSTEM FLUSH LOGS will process all configured logs (the previous code was error-prone and missed a few of them).

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
This is an automated comment for commit 06d45cfa81c2e21963a1a82bb7fabc1bcb9b115b with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/53085/06d45cfa81c2e21963a1a82bb7fabc1bcb9b115b/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Interpreters/Context.cpp [hunk 1]
```diff
@@ -3182,7 +3182,6 @@ void Context::setCluster(const String & cluster_name, const std::shared_ptr<Clus
 
 void Context::initializeSystemLogs()
 {
-    auto lock = getLock();
```
Comment (pufit) [inline src/Interpreters/Context.cpp:3185]: Why was this lock removed?

File: programs/server/config.d/create_logs_at_startup.yaml [hunk 2]
```diff
@@ -0,0 +1 @@
+../../../tests/config/config.d/create_logs_at_startup.yaml
```
Comment (pufit) [inline programs/server/config.d/create_logs_at_startup.yaml:1]: Maybe it wasn't supposed to be committed?

File: src/Interpreters/Context.cpp (hunk 1 above)
Comment (alexey-milovidov) [reply src/Interpreters/Context.cpp:3185]: The function is called once at server initialization and does not require a lock.
But if we use this lock, it leads to a lock order inversion with the initialization of user-defined functions,
which does not lead to anything bad but was detected by TSan.

File: programs/server/config.d/create_logs_at_startup.yaml (hunk 2 above)
Comment (alexey-milovidov) [reply programs/server/config.d/create_logs_at_startup.yaml:1]: No, it is supposed to be committed, 
similarly to other symlinks, it is needed to make local development easier.

**Review by pufit — APPROVED** [review]

File: programs/server/Server.cpp [hunk 3]
```diff
@@ -1650,9 +1650,9 @@ try
         database_catalog.initializeAndLoadTemporaryDatabase();
         loadMetadataSystem(global_context);
         maybeConvertSystemDatabase(global_context);
-        startupSystemTables();
         /// After attaching system databases we can initialize system log.
         global_context->initializeSystemLogs();
+        startupSystemTables();
         global_context->setSystemZooKeeperLogAfterInitializationIfNeeded();
```
Comment (tavplubix) [inline programs/server/Server.cpp:1656]: This change will re-introduce a race condition between the system database initialization and creation of new tables in the database

File: src/Interpreters/Context.cpp (hunk 1 above)
Comment (alexey-milovidov) [reply src/Interpreters/Context.cpp:3166]: I've returned it back with a comment - btw, it's surprisingly convoluted.

**Pull request merged**